apps/checkout-app/pages/api/pay.ts

```diff
@@ -0,0 +1,80 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { createMolliePayment } from "@/backend/payments/providers/mollie";
+import { createOrder } from "@/backend/payments/createOrder";
+import { allowCors } from "@/backend/utils";
+
+export type PaymentProviders = "mollie";
+const paymentProviders: PaymentProviders[] = ["mollie"];
+
+export type Body = {
+  provider: PaymentProviders;
+  checkoutId: string;
+  totalAmount: number;
+  captureAmount?: number; // support for partial payments
+};
+
+export type MollieResponse = {
+  provider: "mollie";
+  data: {
+    checkoutUrl: string;
+  };
+};
+
+export type SuccessResponse = {
+  provider: PaymentProviders;
+  ok: true;
+} & MollieResponse;
+
+export type ErrorResponse = {
+  ok: false;
+  errors: string[];
```

Comment:
Somehwere else we had errors as string[], maybe let's make an Error type even if for now it's string ?

Reply:
I've created specific types for this